### ISSUE TITLE
[Bug] Crash in reader while updating library

### ISSUE THREAD

**Issue by Th3N3rdyGam3r** [CarlosEsco/Neko#283]
### Device information
* Neko version: 2.1.3
* Android phone & version: OnePlus 5T, LineageOS 16.0 (Android 9), MicroG

## Steps to reproduce
1. Run global library update
2. While update is still running, go into browse section
3. Open manga and start reading
4. Crash occured after a short period of time, or only reading a few pages

### Expected behavior
Swiping screen turns to next page in manga

### Actual behavior
Screen freezes and home/app switcher don't work. Locking screen and unlocking unfreezes the home/app switcher. Going to home screen causes android to show that Neko has crashed and should be closed.

### Other details
I changed the logs to Extra and repeated steps to get crash.
Link: https://pastebin.com/UYYLUzXv
I tried reading manga while the library wasn't updating and the crash didn't happen, so I believe that the library updating may have had an influence on the crash.

**Comment by jaredanson** [comment]
Same thing happens to me on the past few versions of the app. I believe it first started when i updated from `2.1.2.1` to `2.1.3`.


I've noticed that if instead of refreshing the whole library, I just refresh a category with a handful of few manga then it works fine. But if the number of manga in a category is quite large then it will crash, just like if trying to refresh the whole library.


Sometimes instead of crashing, it will just timeout/glitch out and not update the manga. I've verified this by trying to do an update, no crash but didn't update anything, then I manually clicked refresh on individual manga and it found new chapters.

**Comment by CarlosEsco** [comment]
Does this happen on the most recent version from yesterday @jaredanson ?  I modified the code some for the 429 error.  So it might have effected this

**Comment by jaredanson** [comment]
@CarlosEsco yes it still happens. I double checked before posting that i was on the newest version, and attempted to refresh the whole library. Doing so the app frozen, and Android prompted me to kill it or wait.

**Comment by jaredanson** [comment]
I haven't checked the logic in the recent releases. Did you modify the number of threads/coroutines or some logic around using them? It feels like app is using up all the resources on the phone when there is a lot of manga in the library.

**Comment by CarlosEsco** [comment]
what device and specs?    Latest version is just using 1 thread i got rid of some of the extra processing that jay originally added to j2k and i had modified some.

**Comment by jaredanson** [comment]
Pixel 2 XL and Android 10, and i'm on the latest version of the app.

**Comment by CarlosEsco** [comment]
how many manga roughly do you have?  How many merged?

**Comment by jaredanson** [comment]
@CarlosEsco what does merged mean?

**Comment by jaredanson** [comment]
And i've got 504 manga in my library +/- a few, i was quickly counting by rows of 3.

**Comment by CarlosEsco** [comment]
thanks.  and in future ungroup the library, long press a manga then press the dot top left to select all 😄

**Comment by jaredanson** [comment]
lmao yeah i figured there was some easy way to do it. But i didn't think it would take that long to count up the rows, but when i got to over 150 rows I was wishing i had looked for the method you mentioned.

**Comment by Th3N3rdyGam3r** [comment]
My library is just short of 2500, so that could be part of the problem.

**Comment by jaredanson** [comment]
@CarlosEsco I noticed another idiosyncrasy about this. If i update a small category and then click to update another small category, which adds to the queue, both of those will pass even though the combined amount is larger than single categories that will fail/freeze up when trying to update.

**Comment by jaredanson** [comment]
I've added 2-3 categories to the queue and it works. When big single categories don't

**Comment by jaredanson** [comment]
@CarlosEsco went manga by manga in the big category seeing if maybe there was some type of issue with a manga entry that was causing the updater to freeze up. That ended up being the case, and luckily it was only one manga so i didn't have to keep searching through the list. When i updated this single manga it caused my phone to lock up and freak out. It was some manga i didn't care about, and after removing it I'm able to update my whole library without any problems.


Obviously going manga by manga trying to figure out which one has an issue is a pain, especially in the case of that other guy who has almost 2,500 entries.

**Comment by jaredanson** [comment]
i initially thought it was some type of thread issue since it was crashing the app and because previous bugs with updating entries just caused an update failure, and not locking up the app/phone. But after fixing it by removing that single manga, it seems to point to some type of an handled exception.

**Comment by CarlosEsco** [comment]
@jaredanson was the manga merged at all ?

**Comment by jaredanson** [comment]
@CarlosEsco i don't know what that means, so i assume no.

**Comment by jaredanson** [comment]
It was some manga I randomly added, I never read or did anything with it.

**Comment by arkon** [comment]
You should probably share what that manga was so it can be debugged.

**Comment by Th3N3rdyGam3r** [comment]
I tracked down the manga that was doing it for me.
I just started a library update in alphabetical order and checked what manga the notification froze on when the app crashed.
[This](https://mangadex.org/title/43789/5toubun-no-hanayome-nino-doujins-doujinshi) is what broke the app for me, even when just updating it by itself. There were a couple failed updates, but only this one would completely crash the app when attempting to update.
Now that I've removed that one, I was able to go through an entire library update without the app crashing, and with a notification actually showing up.
